pkg/client/add_events.go

```diff
@@ -40,6 +42,79 @@ import (
 Wrapper around: https://app.scalyr.com/help/api#addEvents
 */
 
+type EventWithMeta struct {
+	EventBundle *add_events.EventBundle
+	Key         string
+	SessionInfo add_events.SessionInfo
+}
+
+func NewEventWithMeta(
+	bundle *add_events.EventBundle,
+	groupBy []string,
+	serverHost string,
+) EventWithMeta {
+	// initialise
+	key := ""
+	info := make(add_events.SessionInfo)
+
+	// figure out serverHost value
+	usedServerHost := serverHost
+	// if event's ServerHost is set, use it
+	if len(bundle.Event.ServerHost) > 0 {
+		usedServerHost = bundle.Event.ServerHost
+	} else {
+		// if somebody is using library directly and forget to set Event.ServerHost,
+		// lets check the attributes first
+
+		// check serverHost attribute
+		attrHost, ok := bundle.Event.Attrs[add_events.AttrServerHost]
+		if ok {
+			usedServerHost = attrHost.(string)
+		} else {
+			// if serverHost attribute is not set, check the __origServerHost
+			attrOrigHost, okOrig := bundle.Event.Attrs[add_events.AttrOrigServerHost]
+			if okOrig {
+				usedServerHost = attrOrigHost.(string)
+			}
+		}
+	}
+
+	// for the SessionInfo, it has to be in the serverHost
+	// therefore remove the orig and set the serverHost
+	// so that it can be in the next step moved to sessionInfo
+	delete(bundle.Event.Attrs, add_events.AttrOrigServerHost)
+	bundle.Event.Attrs[add_events.AttrServerHost] = usedServerHost
+
+	// construct Key
+	// move attributes from attrs to sessionInfo
+	for _, k := range groupBy {
+		val, ok := bundle.Event.Attrs[k]
+		key += k + ":"
+		if ok {
+			key += fmt.Sprintf("%s", val)
+
+			// move to session info and remove from attributes
+			info[k] = val
+			delete(bundle.Event.Attrs, k)
+		}
+		key += "___DELIM___"
+	}
+
+	// use md5 to shorten the key
+	hash := md5.Sum([]byte(key))
+	bundleKey := hex.EncodeToString(hash[:])
+
+	// during the development we can include the key
+	// so that we can see this in UI
+	// info[add_events.AttrBundleKey] = bundleKey
```

Comment:
This could be useful for debugging so I think it's probably a good idea to add a config option / feature flag (defaults to false) which when set, will make sure this attribute is included.

WDYT?

Reply:
I was border line on this decision. First I was thinking about making it configurable. Then I was thinking, that it's mostly useful during development, so there is no point to make it configurable.

Now, when it's public, maybe some better name then `bundle_key` should be used. I would call it `session_key` - although it's little bit misleading.

WDYT?

Reply:
I have renamed it to `session_key`. :)

Reply:
Thanks for renaming it and adding new `debug` config option.